rust/lance/src/io/commit.rs

```diff
@@ -743,18 +799,24 @@ pub(crate) async fn commit_transaction(
                 let backoff_time = backoff_time(attempt_i);
                 tokio::time::sleep(backoff_time).await;
 
-                let latest_version = dataset.latest_version_id().await?;
-                for version in target_version..=latest_version {
-                    dataset = dataset.checkout_version(version).await?;
-                    let other_transaction = if let Some(txn_file) =
-                        dataset.manifest.transaction_file.as_ref()
-                    {
-                        Some(read_transaction_file(object_store, &dataset.base, txn_file).await?)
-                    } else {
-                        None
-                    };
-                    check_transaction(transaction, version, &other_transaction)?;
-                }
+                dataset.checkout_latest().await?;
+                let latest_version = dataset.manifest.version;
+                futures::stream::iter(target_version..=latest_version)
+                    .map(|version| {
+                        read_dataset_transaction_file(&dataset, version)
+                            .map(move |res| res.map(|tx| (version, tx)))
+                    })
+                    .buffer_unordered(10)
+                    .and_then(|(version, other_transaction)| {
+                        let res = check_transaction(
+                            transaction,
+                            version,
+                            Some(other_transaction.as_ref()),
+                        );
+                        futures::future::ready(res)
+                    })
+                    .try_all(|_| futures::future::ready(true))
```

Comment:
Can you make the `and_then` a `try_for_each`?  Then you don't need the `try_all`?